### ISSUE TITLE
unexpected behavior of neighbour_list

### ISSUE THREAD

**Issue by richardjana** [libAtoms/matscipy#95]
I just encountered behavior from neighbour_list, which I had not expected and which seems strange / undesirable to me when I call the function and give a single number for the cutoff

- When I use a float, e.g. 5.0, everything works just fine.
- When I give an integer, e.g. 5, I get an error `ValueError: object of too small depth for desired array` from _matscipy.neighbour_list.

I realize that the documentation states that you can give "A single float" specifically, but I don't see an obvious reason why "5" should not be interpreted as "5.0".

**Comment by pastewka** [comment]
Interesting, that looks like a bug. Thanks for reporting.

**Comment by jameskermode** [comment]
I've also noticed this before: my guesss is that the variable is passed straight to C without a type conversion so should be pretty easy to fix.

**Comment by pastewka** [comment]
Agree, there is some funkiness going on with type conversion in the C code.

**Comment by pastewka** [comment]
Will be fixed with #96
